src/reconcile/rec_visibility.c

```diff
@@ -427,7 +427,8 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             WT_TIME_WINDOW_SET_START(select_tw, upd);
         else if (select_tw->stop_ts != WT_TS_NONE || select_tw->stop_txn != WT_TXN_NONE) {
             /* If we only have a tombstone in the update list, we must have an ondisk value. */
-            WT_ASSERT(session, vpack != NULL && tombstone != NULL && last_upd->next == NULL);
+            WT_ASSERT(session,
+              vpack != NULL && tombstone != NULL && !vpack->tw.prepare && last_upd->next == NULL);
```

Comment:
Why do you want to check for the prepare flag here?

Reply:
It's checked in the rec_append_orig_value also. I check here it again to emphasise we should not append prepared update to the update chain.